src/specs/networking.md

```diff
@@ -66,3 +68,26 @@ Accepting a `WireTxPayForMessage` into the mempool requires different logic than
+|----------------------------|------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `headerForTxns`            | [Header](./data_structures.md#header)                                                    | Header containing `availableDataRoot`(./data_structures.md#header) of type [HashDigest](./data_structures.md#hashdigest) committing to transaction data.                                                                                              |
+| `headerForISRs`            | [Header](./data_structures.md#header)                                                    | Header containing `availableDataRoot`(./data_structures.md#header) of type [HashDigest](./data_structures.md#hashdigest) committing to intermediate state root data. `lastHeaderHash` within `headerForISRs` equals the hash of `headerForTxns`.|
+| `transactionShareProofs`   | [ShareProof](./data_structures.md#shareproof)`[]`                                        | ShareProof contains both the [Share](./data_structures.md#share) and [NamespaceMerkleTreeInclusionProof](./data_structures.md#namespacemerkletreeinclusionproof) for the Share. `isCol` of type `bool` is set to `false`.                                           |
+| `stateShareProofs`         | [ShareProof](./data_structures.md#shareproof)`[]`                                        | ShareProof contains both the [Share](./data_structures.md#share) and [NamespaceMerkleTreeInclusionProof](./data_structures.md#namespacemerkletreeinclusionproof) for the Share. `isCol` of type `bool` is set to `false`.                                           |
+| `wrapIndex`                | `uint64`                                                                                 | Index for connecting the [WrappedIntermediateStateRoot](./data_structures.md#wrappedintermediatestateroot) and [WrappedTransaction](./data_structures.md#wrappedtransaction) after shares are parsed.                                                                                      |
+| `intermediateStateElements`| [(key,value)](./data_structures.md#state)                                                | Keys are of type `byte[32]` and values are byte arrays.                                                                                                                                                                                                                                      |
```

Comment:
I would define a new struct here for state elements, otherwise it's not obvious how to serialize it.

Reply:
A new struct for state elements defined in data_structures.md.